### PR TITLE
Inherit `color` behavior from `CARGO_TERM_COLOR` env var

### PR DESCRIPTION
**Pull request by max-sixty**
ref https://doc.rust-lang.org/cargo/reference/config.html#termcolor

### CODE REVIEW

**Comment by mitsuhiko** [comment]
I like handling this, but I'm not sure if this is the place to put it. I'm okay with this if it does not show up in the help output and I'm not sure if `env` does.

**Comment by max-sixty** [comment]
It does show up...

```
cargo-insta 1.28.0
A helper utility to work with insta snapshots

USAGE:
    cargo insta <SUBCOMMAND>

OPTIONS:
        --color <WHEN>    Coloring [env: CARGO_TERM_COLOR=]  [default: auto]  [possible values: auto, always, never]
    -h, --help            Prints help information
    -V, --version         Prints version information
```

Where would be better? To what extent could this thought of as "inherited from cargo" in a similar way to `--lib` etc?

**Comment by mitsuhiko** [comment]
The easiest would be to change it into an `Option` and then handle the `None` case explicitly in `handle_color`.

**Comment by max-sixty** [comment]
@mitsuhiko lmk if that's what you were thinking...

File: cargo-insta/src/cli.rs [hunk 1]
```diff
@@ -304,12 +304,21 @@ fn query_snapshot(
...
-        "auto" => {}
-        "never" => set_colors_enabled(false),
-        color => return Err(err_msg(format!("invalid value for --color: {}", color))),
+        None => match env::var("CARGO_TERM_COLOR") {
+            Ok(v) => handle_color(Some(v.as_str())).or_else(|_| {
+                return Err(err_msg(format!(
+                    "invalid value for CARGO_TERM_COLOR: {}",
+                    color.unwrap()
+                )));
+            })?,
```
Comment (max-sixty) [inline cargo-insta/src/cli.rs:315]: Open to feedback on whether this is good rust...

**Comment by mitsuhiko** [comment]
Will do this differently.

**Pull request closed without merging**